src/Commands/pm/SecurityUpdateCommands.php

```diff
@@ -112,4 +78,151 @@ public function suggestComposerCommand($result, CommandData $commandData)
+        try {
+            $response_body = file_get_contents('https://raw.githubusercontent.com/drupal-composer/drupal-security-advisories/8.x/composer.json');
+        } catch (Exception $e) {
+            throw new Exception("Unable to fetch drupal-security-advisories information.");
+        }
+        $security_advisories_composer_json = json_decode($response_body, TRUE);
+        return $security_advisories_composer_json;
+    }
+
+    /**
+     * Loads the contents of the local Drupal application's composer.lock file.
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    protected function loadSiteComposerLock() {
+        $composer_root = Drush::bootstrapManager()->getComposerRoot();
+        $composer_lock_file_name = getenv('COMPOSER') ? str_replace('.json', '',
+            getenv('COMPOSER')) : 'composer';
+        $composer_lock_file_name .= '.lock';
+        $composer_lock_file_path = Path::join($composer_root,
+            $composer_lock_file_name);
+        if (!file_exists($composer_lock_file_path)) {
+            throw new Exception("Cannot find $composer_lock_file_path!");
+        }
+        $composer_lock_contents = file_get_contents($composer_lock_file_path);
+        $composer_lock_data = json_decode($composer_lock_contents, TRUE);
+        if (!array_key_exists('packages', $composer_lock_data)) {
+            throw new Exception("No packages were found in $composer_lock_file_path! Contents:\n $composer_lock_contents");
+        }
+        return $composer_lock_data;
+    }
+
+    /**
+     * Register all available security updates in $this->securityUpdates.
+     * @param array $composer_lock_data
+     *   The contents of the local Drupal application's composer.lock file.
+     * @param array $security_advisories_composer_json
+     *   The composer.json array from drupal-security-advisories.
+     */
+    protected function registerAllSecurityUpdates(
+        $composer_lock_data,
+        $security_advisories_composer_json
+    ) {
+        foreach ($composer_lock_data['packages'] as $key => $package) {
+            $name = $package['name'];
+            $this->registerPackageSecurityUpdates($security_advisories_composer_json, $name, $package);
+        }
+    }
+
+    /**
+     * Determines if update is avaiable based on a conflict constraint.
+     *
+     * @param string $conflict_constraint
+     *   The constraint for the conflicting, insecure package version.
+     *   E.g., <1.0.0.
+     * @param array $package
+     *   The package to be evaluated.
+     * @param string $name
+     *   The human readable display name for the package.
+     *
+     * @return array
+     *   An associative array containing name, version, and min-version keys.
+     */
+    public static function determineUpdatesFromConstraint(
+        $conflict_constraint,
+        $package,
+        $name
+    ) {
+        // Only parse constraints that follow pattern like "<1.0.0".
```

Comment:
If both `<8.5.1` and `8.5.0` conflict constraints are defined in drupal-security-advisories, it may cause multiple available updates to be registered. We need to somehow sort the list of conflicts such that the most relevant one is evaluated first, then we need to return and stop evaluating subsequent constraints.